tests/unittests/test-charcode.cpp

```diff
@@ -112,3 +172,188 @@ TEST_F(CharWidthCache, FontNo)
 	EXPECT_EQ(WCODE::GetFontNo(L'あ'), 1);
 	EXPECT_EQ(WCODE::GetFontNo2(0xd83c, 0xdf38), 1);
 }
+
+TEST(charcode, IS_KEYWORD_CHAR)
+{
+	for (wchar_t ch = 0; ch < _countof(gm_keyword_char); ++ch) {
+		EXPECT_EQ(IS_KEYWORD_CHAR(ch),
+			gm_keyword_char[ch] == CK_CSYM || gm_keyword_char[ch] == CK_UDEF);
+	}
+}
+
+// 以下、関数が判定している文字がすべてASCII範囲内であれば、テーブルを使って総当たりする。
+// ASCII範囲外の文字が含まれる関数については適宜テストケースを手書きして対応。
+
+TEST(charcode, IsAZ)
+{
+	constexpr std::array<bool, 128> table = {
+		false, false, false, false, false, false, false, false, false, false,
+		false, false, false, false, false, false, false, false, false, false,
+		false, false, false, false, false, false, false, false, false, false,
+		false, false, false, false, false, false, false, false, false, false,
+		false, false, false, false, false, false, false, false, false, false,
+		false, false, false, false, false, false, false, false, false, false,
+		false, false, false, false, false, true,  true,  true,  true,  true,
+		true,  true,  true,  true,  true,  true,  true,  true,  true,  true,
+		true,  true,  true,  true,  true,  true,  true,  true,  true,  true,
+		true,  false, false, false, false, false, false, true,  true,  true,
+		true,  true,  true,  true,  true,  true,  true,  true,  true,  true,
+		true,  true,  true,  true,  true,  true,  true,  true,  true,  true,
+		true,  true,  true,  false, false, false, false, false
+	};
+	for (wchar_t ch = 0; ch < table.size(); ++ch) {
+		EXPECT_EQ(WCODE::IsAZ(ch), table[ch]);
```

Comment:
最新の定義は(expect,actual)じゃなくなってるんですね。

```
# define ASSERT_EQ(val1, val2) GTEST_ASSERT_EQ(val1, val2)
```

Reply:
(actual, expected)だと書いてあります。実際の定義を見る限りあまり厳密に区別してない印象は受けますが。
> However, when possible, ASSERT_EQ(actual, expected) is preferred to ASSERT_TRUE(actual == expected), since it tells you actual and expected's values on failure.

https://github.com/google/googletest/blob/master/docs/primer.md

Reply:
なんでココにこだわるかというと、過去にそういう指摘をしてきたからです。

googletestはもともとjavaのjunitを参考に作られているので、
基本的にはjunitのテストアサーションと似た感じになっています。

よくよく考えてみるとC++の標準的な呼出し規約だと関数の引数は右から左に解釈されるので、`(actual, expected)`の方が適切な気もします。

理由もなくポリシー変えるのは違うと思うのであえて食いついてみた感じです。